instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3OutputStreamWriteBytesAndOffsetAdvice.java

```diff
@@ -0,0 +1,28 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+package io.opentelemetry.javaagent.instrumentation.servlet.v3_0;
+
+import static io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet.Injection.getInjectionObject;
+
+import io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet.InjectedInfo;
+import io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet.InjectionObject;
+import java.io.UnsupportedEncodingException;
+import javax.servlet.ServletOutputStream;
+import net.bytebuddy.asm.Advice;
+
+public class Servlet3OutputStreamWriteBytesAndOffsetAdvice {
+
+  @Advice.OnMethodEnter(suppress = Throwable.class)
+  public static void methodEnter(
+      @Advice.This ServletOutputStream servletOutputStream,
+      @Advice.Argument(value = 0, readOnly = false) byte[] write,
+      @Advice.Argument(1) int off,
+      @Advice.Argument(value = 2, readOnly = false) int len)
+      throws UnsupportedEncodingException {
+    InjectionObject obj = getInjectionObject(servletOutputStream);
+    InjectedInfo info = obj.stringInjection(write, off, len);
+    if (info != null) {
+      write = info.bits;
+      len = info.length;
```

Comment:
could we simplify / make something like this work?

```suggestion
      write = info.bits;
      off = 0;
      len = bits.length;
```